### PR TITLE
[flutter_tools] add --android-gradle-daemon option, use in devicelab

### PR DESCRIPTION
**Pull request by jonahwilliams**
## Description

Remove devicelab specific code for shutting down gradle daemon, add `--android-gradle-daemon` option to build/run/drive`. Avoids need for un-tested devicelab specific handler. There are also some feature requests for this, so 2 birds one stone.

Example:

`flutter build apk --no-android-gradle-daemon` will pass `--no-daemon` on to gradle

### CODE REVIEW

File: packages/flutter_tools/lib/src/build_info.dart [hunk 1]
```diff
@@ -118,6 +119,20 @@ class BuildInfo {
   /// will be written for code size profiling.
   final String codeSizeDirectory;
 
+  /// Whether to enable the Gradle daemon when performing an Android build.
+  ///
+  /// Starting the daemon is the default behavior of the gradle wrapper script created
```
Comment (zanderso) [inline packages/flutter_tools/lib/src/build_info.dart:124]: Should this go in the help text for the flag?

File: dev/devicelab/lib/framework/runner.dart [hunk 2]
```diff
@@ -88,7 +87,6 @@ Future<TaskResult> runTask(
   } finally {
     if (!runnerFinished)
       runner.kill(ProcessSignal.sigkill);
-    await cleanupSystem();
```
Comment (zanderso) [inline dev/devicelab/lib/framework/runner.dart:91]: Is there any way to check that the gradle daemon is not running?

**Review by zanderso — APPROVED** [review]

File: dev/devicelab/lib/framework/runner.dart (hunk 2 above)
Comment (jonahwilliams) [reply dev/devicelab/lib/framework/runner.dart:91]: We could look for still running java processes in the before/after process check

File: packages/flutter_tools/lib/src/build_info.dart (hunk 1 above)
Comment (jonahwilliams) [reply packages/flutter_tools/lib/src/build_info.dart:124]: Done

**Comment by jonahwilliams** [comment]
iOS/macOS failures is the current SDK flake and is unrelated

**Comment by whesse** [comment]
This was relanded in https://github.com/flutter/flutter/pull/68491

**Pull request merged**